test/unit/org/apache/cassandra/schema/SchemaStatementWarningsTest.java

```diff
@@ -0,0 +1,126 @@
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaStatementWarningsTest.addWarnToList($!)"),
+                       @BMRule(name = "client warning 3",
+                               targetClass = "AlterSchemaStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaStatementWarningsTest.addWarnToList($!)") })
+    public void testClientWarningsOnCreateKeyspace() throws Throwable
+    {
+        ClientWarn.instance.captureWarnings();
+        injectedWarnings.clear();
+        createKeyspace("CREATE KEYSPACE %s WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}");
+
+        Assertions.assertThat(injectedWarnings).contains(msg1, msg2); // failure here means the bm rules need to be updated
+        Assertions.assertThat(ClientWarn.instance.getWarnings()).containsExactlyInAnyOrder(msg1, msg2);
+    }
+
+    @Test
+    @BMRules(rules = { @BMRule(name = "client warning 1",
+                               targetClass = "CreateTableStatement",
+                               targetMethod = "apply",
+                               targetLocation = "AT INVOKE TableMetadata.validate",
+                               action = "org.apache.cassandra.schema.SchemaStatementWarningsTest.addWarn()"),
+                       @BMRule(name = "client warning 2",
+                               targetClass = "CreateTableStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaStatementWarningsTest.addWarnToList($!)"),
+                       @BMRule(name = "client warning 3",
+                               targetClass = "AlterSchemaStatement",
+                               targetMethod = "clientWarnings",
+                               targetLocation = "AT EXIT",
+                               action = "return org.apache.cassandra.schema.SchemaStatementWarningsTest.addWarnToList($!)"),
+    })
+    public void testClientWarningsOnCreateTable() throws Throwable
+    {
+        ClientWarn.instance.captureWarnings();
+        injectedWarnings.clear();
+        createTable("CREATE TABLE %s (k int primary key, v int)");
+
+        Assertions.assertThat(injectedWarnings).contains(msg1, msg2); // failure here means the bm rules need to be updated
+        Assertions.assertThat(ClientWarn.instance.getWarnings()).containsExactlyInAnyOrder(msg1, msg2);
+    }
+
+    static volatile String msg1, msg2;
+
+    @Before
+    public void before()
+    {
+        msg1 = UUID.randomUUID().toString();
+        msg2 = UUID.randomUUID().toString();
+    }
+
+    static Queue<String> injectedWarnings = new ConcurrentLinkedQueue<String>();
```

Comment:
This can be `private`. Also we could place it at the beginning of the class, before methods. We don't need the second `<String>`, so it can be:
```java
private static final Queue<String> injectedWarnings = new ConcurrentLinkedQueue<>();
```